gallery/src/components/NxDateInput/NxDateInputSimpleExample.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState } from 'react';
+
+import {
+  NxDateInput,
+  NxDateInputStateProps,
+  nxTextInputStateHelpers
+} from '@sonatype/react-shared-components';
+
+const { initialState, userInput } = nxTextInputStateHelpers;
+
+export default function NxTextInputSimpleExample() {
+  // Note that NxTextInputStateProps is the TypeScript type for the state helper function return types.
+  // The explicity type parameter isn't necessary here, it's only present to demonstrate that type.
+  const [state, setState] = useState<NxDateInputStateProps>(initialState(''));
+
+  function onChange(val: string) {
+    setState(userInput(null, val));
+  }
+
+  return (
+    <NxDateInput { ...state } onChange={onChange} />
+  );
+}
```

Comment:
I think another example with some validation would be good (maybe validate that the date is non-empty and more recent than 1900 or something).

Reply:
> I think another example with some validation would be good (maybe validate that the date is non-empty and more recent than 1900 or something).

Added: 76e109d